__test__/responses/single-vault.json

```diff
@@ -1,11 +0,0 @@
-{
-  "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
```

Comment:
I see that we no longer need this JSON in our tests, but now it feels that the test is incomplete. 

I would consider the test to also see if the vault response is properly interpreted by the SDK (i.e. all fields are populated properly). This may give more confidence on the SDK's functionality in general and it can also help us quickly identify when something goes wrong (wither Connect changed the response or the SDK misses something).

What do you think?

Reply:
Good suggestion. But I think we may not need it at all, as in the current tests implementation we do not check whether each field is populated. We only check that returned object is `instance of` some class.

For my opinion there should be a separate test-case for `ObjectSerializer` that checks exactly `deserialize`/`serealize` methods, and make sure that it returns exactly what we expect (class, populated fields).

Reply:
Sounds like a plan then. We will add these test cases with a different PR, since that's out of this PR's scope.